src/Compilers/Shared/BuildClient.cs

```diff
@@ -249,6 +255,30 @@ private static bool UseNativeArguments()
             return true;
         }
 
+        private static bool AreNamedPipesSupported()
+        {
+            if (!PlatformInformation.IsRunningOnMono)
+                return true;
+
+            IDisposable npcs = null;
+            try 
+            {
+                // Mono configurations without named pipe support will throw a PNSE at some point in this process.
+                npcs = new NamedPipeClientStream(".", "nonexistentpipe", PipeDirection.InOut);
+                npcs.Dispose();
+                return true;
+            } 
+            catch (PlatformNotSupportedException) 
+            {
+                if (npcs != null) 
+                {
+                    // Compensate for broken finalizer in older builds of mono
```

Comment:
Can you link to a bug here?

Reply:
We never filed a bug, but the fix & explanation are in https://github.com/mono/mono/pull/12322/commits/2a731f29b065392ca9b44d6613abee2aa413a144

Reply:
Lets link to that then. At least gives context to the comment. 

---
In reply to: [253139222](https://github.com/dotnet/roslyn/pull/31497#discussion_r253139222) [](ancestors = 253139222)